### DISCUSSION TITLE
New Source: AWS Cost Explorer

### DISCUSSION THREAD

**Discussion by miguel-firebolt** [airbytehq/airbyte#33147] [New Connector Request]
## Tell us about the new connector you’d like to have
* Which source and which destination? new source for AWS Cost Explorer [API](https://docs.aws.amazon.com/aws-cost-management/latest/APIReference/API_Operations_AWS_Cost_Explorer_Service.html) [boto3](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ce.html)
* Do you need a specific version of the underlying data source e.g: you specifically need support for an older version of the API or DB? No

## Describe the context around this new connector
* Why do you need this integration? How does your team intend to use the data? This helps us understand the use case. We want to build reports showing AWS costs by service and account over time
* How often do you want to run syncs? hourly
* If this is an API source connector, which entities/endpoints do you need supported? GetCostAndUsage
* If the connector is for a paid service, can we name you as a mutual user when we subscribe for an account? Which company should we name? 

## Describe the alternative you are considering or using
What are you considering doing if you don’t have this integration through Airbyte? Schedule cost explorer report

## Are you willing to submit a PR?
Possibly

**Comment by sherifnada** [comment]
hey @miguel-firebolt , really interesting idea! thanks for creating the issue. we're currently focused on reliability and performance of existing connectors, so the fastest way to get this connector is via a PR. Otherwise stay tuned on this ticket for updates.

**Comment by miguel-firebolt** [comment]
Thanks @sherifnada. Understood, I will see if we have the bandwidth to start on this connector and update here if we plan to work on it.

**Comment by eskarimov** [comment]
Hi @miguel-firebolt, the idea is indeed very nice, will be useful for many teams. 

I'd like to try contributing to it, and I've got a question to clarify desired functionality:

In the PR description the endpoint is specified as `GetCostAndUsageWithResources`. According to [AWS docs](https://docs.aws.amazon.com/aws-cost-management/latest/APIReference/API_GetCostAndUsageWithResources.html) this API is available only for EC2 services. So I'm curious if we should use this API or [`GetCostAndUsage`](https://docs.aws.amazon.com/aws-cost-management/latest/APIReference/API_GetCostAndUsage.html) instead, in order to get info about other services?

**Comment by miguel-firebolt** [comment]
@eskarimov -- good point! I agree GetCostAndUsage is probably going to be a better endpoint to start with.

[a comment by an automated account is not shown]
